### ISSUE TITLE
flash arm64 image for Raspberry Pi 3

### ISSUE THREAD

**Issue by tombelv** [duckietown/duckietown-shell#74]
I'm not sure if this is the right place to ask, but I have recently started working with some new duckiebots (DB19) and I'm looking for a way to install a 64 bit version of the duckieOS to implement my algorithms (since I'm working with a 64bit-only library).

I have found some references to this in the [duckietown-shell-commands](https://github.com/duckietown/duckietown-shell-commands) repository but from [dtproject_utils.py](https://github.com/duckietown/duckietown-shell-commands/blob/daffy/utils/dtproject_utils.py) it seems that it is currently not supported.

Circumventing the duckietown-shell, I have also installed a 64bit version of HypriotOS on my Raspberrry Pi but then I'm not sure of how I should proceed to achieve a similar configuration.

Do you have any pointers? I would gladly contribute if there is something I can help with.

**Comment by CourchesneA** [comment]
As far as I know, this is not supported yet. We have parallel work to enable our workflows on the Jetson Nano which is running arm64v8 images so you could re-use a lot of this stuff. @afdaniele Do you know which part of the jetson nano setup would work on a raspberry pi ?

Depending on what library you are trying to use, for some workflow we run the containers on a laptop and communicate with the duckiebot remotely. For instance, if your library is used for control, you can wrap it up in a ROS node and run it from your laptop, that would definitely be the easiest way

**Comment by tombelv** [comment]
> Depending on what library you are trying to use, for some workflow we run the containers on a laptop and communicate with the duckiebot remotely. For instance, if your library is used for control, you can wrap it up in a ROS node and run it from your laptop, that would definitely be the easiest way

Yes that could be ta temporary workaround, although my final goal is to perform all the computations onboard.

Since you said "yet", does that mean that you are planning to make 64bit images possible also for the raspberry pi?
I guess that since most (all?) of your software is written in Python it should "just" be a matter of configuration, if the dependancies are there (but I guess that they are if the jetson nano works).

Unfortunately I'm a novice in both Docker and ROS so I may be missing something, but since it seems like the arm64v8 images are available, can you tell me what would happen if I manually installed [the containers mentioned here](https://docs.duckietown.org/daffy/duckietown-robotics-development/out/duckietown_code_structure.html) on a 64bit HypriotOS image?

**Comment by CourchesneA** [comment]
This was not explicitely planned but more of a "nice to have" feature. In theory, you could modify the arch in this file:
https://github.com/duckietown/duckietown-shell-commands/blob/daffy/disk_image/create/raspberry_pi/disk_template/root/data/config/autoboot/duckiebot.yaml

and then run it using docker-compose to start all the images (make sure they are up to date with `docker pull` before). In practice, this has not been tested so I would assume that would not work. There are some stuff that is being done in `init_sd_card` and in the image that we ship that you will not have. I am not familiar with that that is, but you might have to set your hostname and avahi manually

**Comment by tombelv** [comment]
Ok I will try looking at what happens by doing this.

Also, while we wait for @afdaniele (if he happens to have some advice to give), do you know how can I make `init_sd_card` pull my 64bit HypriotOS image instead of the default one?
I've tried changing [this line](https://github.com/duckietown/duckietown-shell-commands/blob/fc42baed0cdc0362656c230c7a0cb58414de9eab/disk_image/create/raspberry_pi/private_command.py#L52) to point to the 64bit one but it does not seem to do anything so I must be misundertanding something. I know that it is not enough to get what I want but it might be a start.
